### PR TITLE
Settlement component sync

### LINKED ISSUE #736: SettlementComponent Sync
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Sync the following properties
- Owner
- Gold
- IsOwnerUnassigned

## Intended Design

Create necessary files in the GameInterface.Services.Settlement namespace/folder structure

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

N/A

## Requirements

[HLR5](https://docs.google.com/document/d/1SAIUng44uFV3xKxJce7II9gGVJwOwvbyNkmYKxY-o8A/edit?usp=sharing)
[HLR6](https://docs.google.com/document/d/1SAIUng44uFV3xKxJce7II9gGVJwOwvbyNkmYKxY-o8A/edit?usp=sharing)
[HLR7](https://docs.google.com/document/d/1SAIUng44uFV3xKxJce7II9gGVJwOwvbyNkmYKxY-o8A/edit?usp=sharing)

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by MaxDorob**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the new behaviour?

Resolves   #736

### CODE REVIEW

File: source/Coop.Core/Client/Services/Settlements/Handlers/ClientSettlementComponetHandler.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+{
+    /// <summary>
+    /// Handles <see cref="NetworkSettlementComponentChangedGold"/>, <see cref="NetworkSettlementComponentChangedIsOwnerUnassigned"/> and <see cref="NetworkSettlementComponentChangedOwner"/> from the server and then goes to the <see cref="GameInterface.Services.Settlements.Handlers.SettlementComponentHandler"/>.
+    /// </summary>
+    public class ClientSettlementComponetHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+
+        public ClientSettlementComponetHandler(IMessageBroker messageBroker, INetwork network)
```
Comment (samuelzurowski) [inline source/Coop.Core/Client/Services/Settlements/Handlers/ClientSettlementComponetHandler.cs]: Please Rename class & file to `ClientSettlementComponentHandler`

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Gold"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedGold : IEvent
+    {
+        [ProtoMember(1)]
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs]: Properties should not have set

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        [ProtoMember(2)]
+        public int Gold { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs]: remove set

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.IsOwnerUnassigned"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedIsOwnerUnassigned : IEvent
+    {
+        [ProtoMember(1)]
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs]: remove set

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        [ProtoMember(2)]
+        public bool IsOwnerUnassigned { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs]: remove set

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Owner"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedOwner : IEvent
+    {
+        [ProtoMember(1)]
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs]: remove set

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        [ProtoMember(2)]
+        public string OwnerId { get; set; }
```
Comment (samuelzurowski) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentGold.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Let the client know to change <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Gold"/>
+    /// </summary>
+    public record ChangeSettlementComponentGold : ICommand
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentGold.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentGold.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        public int Gold { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentGold.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentIsOwnerUnassigned.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Let the client know to change <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.IsOwnerUnassigned"/>
+    /// </summary>
+    public record ChangeSettlementComponentIsOwnerUnassigned : ICommand
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentIsOwnerUnassigned.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentIsOwnerUnassigned.cs [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        public bool IsOwnerUnassigned { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentIsOwnerUnassigned.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentOwner.cs [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Let the client know to change <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Owner"/>
+    /// </summary>
+    public record ChangeSettlementComponentOwner : ICommand
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentOwner.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentOwner.cs [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+        public string OwnerId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/ChangeSettlementComponentOwner.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Gold"/> changed
+    /// </summary>
+    public record SettlementComponentChangedGold : IEvent
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+        public int Gold { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.IsOwnerUnassigned"/> changed
+    /// </summary>
+    public record SettlementComponentChangedIsOwnerUnassigned : IEvent
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+        public bool IsOwnerUnassigned { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Owner"/> changed
+    /// </summary>
+    public record SettlementComponentChangedOwner : IEvent
+    {
+        public string SettlementComponentId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs]: remove set

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+        public string OwnerId { get; set; }
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs]: remove set

File: source/GameInterface/Services/Settlements/Patches/OwnerSettlementComponentPatch.cs [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace GameInterface.Services.Settlements.Patches
+{
+    /// <summary>
+    /// Patch for <see cref="SettlementComponent.Owner"/>
+    /// </summary>
+    [HarmonyPatch(typeof(SettlementComponent))]
+    public static class OwnerSettlementComponentPatch
+    {
+        private static ILogger Logger = LogManager.GetLogger<Settlement>();
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Patches/OwnerSettlementComponentPatch.cs]:
```c#
private static ILogger Logger = LogManager.GetLogger<OwnerSettlementComponentPatch>();
```

File: source/GameInterface/Services/Settlements/Patches/IsOwnerUnassignedSettlementComponentPatch.cs [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace GameInterface.Services.Settlements.Patches
+{
+    /// <summary>
+    /// Patch for <see cref="SettlementComponent.IsOwnerUnassigned"/>
+    /// </summary>
+    [HarmonyPatch(typeof(SettlementComponent))]
+    public static class IsOwnerUnassignedSettlementComponentPatch
+    {
+        private static ILogger Logger = LogManager.GetLogger<Settlement>();
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Patches/IsOwnerUnassignedSettlementComponentPatch.cs]:
```c#
private static ILogger Logger = LogManager.GetLogger<IsOwnerUnassignedSettlementComponentPatch>();
```

File: source/GameInterface/Services/Settlements/Patches/GoldSettlementComponentPatch.cs [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace GameInterface.Services.Settlements.Patches
+{
+    /// <summary>
+    /// Patch for <see cref="SettlementComponent.Gold"/>
+    /// </summary>
+    [HarmonyPatch(typeof(SettlementComponent))]
+    public static class GoldSettlementComponentPatch
+    {
+        private static ILogger Logger = LogManager.GetLogger<Settlement>();
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Patches/GoldSettlementComponentPatch.cs]:
```c#
private static ILogger Logger = LogManager.GetLogger<GoldSettlementComponentPatch>();
```

File: source/GameInterface/Extentions/SettlementComponentExtensions.cs [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace GameInterface.Extentions
+{
+    /// <summary>
+    /// <see cref="SettlementComponent"/> setter accessors
+    /// </summary>
+    public static class SettlementComponentExtensions
+    {
+        static PropertyInfo GoldProperty = typeof(SettlementComponent).GetProperty(nameof(SettlementComponent.Gold));
+        public static void SetGold(this SettlementComponent obj, int value) => GoldProperty.SetValue(obj, value);
```
Comment (samuelzurowski) [inline source/GameInterface/Extentions/SettlementComponentExtensions.cs]: soon this wont be needed due to publicize.

File: source/GameInterface/Services/Settlements/Commands/SettlementCommands.cs [hunk 24]
```diff
@@ -434,4 +434,122 @@ public static string Info(List<string> args)
...
+        string settlementComponentId = args[0];
+        if (!int.TryParse(args[1], out int gold))
+        {
+            return "Unable to parse gold amount";
+        }
+        if (objectManager.TryGetObject<SettlementComponent>(settlementComponentId, out var settlementComponent) == false)
+            return $"SettlementComponent: {settlementComponentId} was not found.";
+
+        try
+        {
```
Comment (samuelzurowski) [inline source/GameInterface/Services/Settlements/Commands/SettlementCommands.cs]: has these had exceptions come up? if it works a try isnt needed.

Comment (samuelzurowski) [reply source/GameInterface/Services/Settlements/Commands/SettlementCommands.cs]: If so remove all try/catches

**Review by samuelzurowski — CHANGES_REQUESTED** [review]
please fix these changes. mostly minor issues.

File: source/GameInterface/Extentions/SettlementComponentExtensions.cs (hunk 23 above)
Comment (samuelzurowski) [reply source/GameInterface/Extentions/SettlementComponentExtensions.cs]: for now keep of course.

**Review by samuelzurowski — DISMISSED** [review]
looks good to me @garrettluskey  please verify

File: source/GameInterface/Services/Settlements/Patches/GoldSettlementComponentPatch.cs [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+        [HarmonyPatch(nameof(SettlementComponent.Gold), MethodType.Setter)]
+        public static bool Prefix(SettlementComponent __instance, int value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client created unmanaged {name}\n"
+                    + "Callstack: {callstack}", typeof(SettlementComponent), Environment.StackTrace);
+                return true;
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Patches/GoldSettlementComponentPatch.cs]: Return false for client instead

```suggestion
                return false;
```

File: source/GameInterface/Services/Settlements/Patches/IsOwnerUnassignedSettlementComponentPatch.cs [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+        [HarmonyPatch(nameof(SettlementComponent.IsOwnerUnassigned), MethodType.Setter)]
+        public static bool Prefix(SettlementComponent __instance, bool value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client created unmanaged {name}\n"
+                    + "Callstack: {callstack}", typeof(SettlementComponent), Environment.StackTrace);
+                return true;
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Patches/IsOwnerUnassignedSettlementComponentPatch.cs]: Return false for client instead

```suggestion
                return false;
```

File: source/GameInterface/Services/Settlements/Patches/OwnerSettlementComponentPatch.cs [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+        [HarmonyPatch(nameof(SettlementComponent.Owner), MethodType.Setter)]
+        public static bool Prefix(SettlementComponent __instance, PartyBase value)
+        {
+            if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+            if (ModInformation.IsClient)
+            {
+                Logger.Error("Client created unmanaged {name}\n"
+                    + "Callstack: {callstack}", typeof(SettlementComponent), Environment.StackTrace);
+                return true;
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Patches/OwnerSettlementComponentPatch.cs]: Return false for client instead

```suggestion
                return false;
```

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Owner"/> changed
+    /// </summary>
+    public record SettlementComponentChangedOwner : IEvent
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedOwner.cs]:
```suggestion
    public record SettlementComponentOwnerChanged : IEvent
```

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs [hunk 29]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.IsOwnerUnassigned"/> changed
+    /// </summary>
+    public record SettlementComponentChangedIsOwnerUnassigned : IEvent
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedIsOwnerUnassigned.cs]:
```suggestion
    public record SettlementComponentIsOwnerUnassignedChanged : IEvent
```

File: source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs [hunk 30]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Gold"/> changed
+    /// </summary>
+    public record SettlementComponentChangedGold : IEvent
```
Comment (garrettluskey) [inline source/GameInterface/Services/Settlements/Messages/SettlementComponentChangedGold.cs]:
```suggestion
    public record SettlementComponentGoldChanged : IEvent
```

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Owner"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedOwner : IEvent
```
Comment (garrettluskey) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedOwner.cs]:
```suggestion
    public record NetworkChangeSettlementComponentOwner : ICommand
```

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs [hunk 32]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.IsOwnerUnassigned"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedIsOwnerUnassigned : IEvent
```
Comment (garrettluskey) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedIsOwnerUnassigned.cs]:
```suggestion
    public record NetworkChangeSettlementComponentIsOwnerUnassigned : ICommand
```

File: source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs [hunk 33]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace GameInterface.Services.Settlements.Messages
+{
+    /// <summary>
+    /// Notify clients of <see cref="TaleWorlds.CampaignSystem.Settlements.SettlementComponent.Gold"/> set
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public record NetworkSettlementComponentChangedGold : IEvent
```
Comment (garrettluskey) [inline source/Coop.Core/Server/Services/Settlements/Messages/NetworkSettlementComponentChangedGold.cs]:
```suggestion
    public record NetworkChangeSettlementComponentGold : ICommand
```

**Review by garrettluskey — CHANGES_REQUESTED** [review]

**Review by garrettluskey — APPROVED** [review]

**Pull request merged**